terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/runners.ts

```diff
@@ -440,6 +458,206 @@ async function getCreateRunnerSubnetSequence(
+    }
+    try {
+      if (runnerParameters.orgName !== undefined) {
+        metrics.runnersReuseTryOrg(
+          runners.length,
+          runnerParameters.orgName,
+          runnerParameters.runnerType.runnerTypeName,
+        );
+      } else if (runnerParameters.repoName !== undefined) {
+        metrics.runnersReuseTryRepo(
+          runners.length,
+          getRepo(runnerParameters.repoName),
+          runnerParameters.runnerType.runnerTypeName,
+        );
+      }
+
+      await redisLocked(
+        `tryReuseRunner`,
+        runner.instanceId,
+        async () => {
+          // I suspect it will be too many requests against GH API to check if runner is really offline
+
+          if (ssmM.has(runner.awsRegion) === false) {
+            ssmM.set(runner.awsRegion, new SSM({ region: runner.awsRegion }));
+          }
+          const ssm = ssmM.get(runner.awsRegion) as SSM;
+          if (ec2M.has(runner.awsRegion) === false) {
+            ec2M.set(runner.awsRegion, new EC2({ region: runner.awsRegion }));
+          }
+          const ec2 = ec2M.get(runner.awsRegion) as EC2;
+
+          // should come before removing other tags, this is useful so
+          // there is always a tag present for scaeDown to know that
+          // it can/will be reused and avoid deleting it
+          await expBackOff(() => {
+            return metrics.trackRequestRegion(
+              runner.awsRegion,
+              metrics.ec2CreateTagsAWSCallSuccess,
+              metrics.ec2CreateTagsAWSCallFailure,
+              () => {
+                return ec2
+                  .createTags({
+                    Resources: [runner.instanceId],
+                    Tags: [{ Key: 'EBSVolumeReplacementRequestTm', Value: `${Math.floor(+new Date() / 1000)}` }],
+                  })
+                  .promise();
+              },
+            );
+          });
+          console.debug(`[tryReuseRunner]: Reuse of runner ${runner.instanceId}: Created reuse tag`);
+
+          await expBackOff(() => {
+            return metrics.trackRequestRegion(
+              runner.awsRegion,
+              metrics.ec2DeleteTagsAWSCallSuccess,
+              metrics.ec2DeleteTagsAWSCallFailure,
+              () => {
+                return ec2
+                  .deleteTags({
+                    Resources: [runner.instanceId],
+                    Tags: [{ Key: 'GithubRunnerID' }, { Key: 'EphemeralRunnerFinished' }],
+                  })
+                  .promise();
+              },
+            );
+          });
+          console.debug(`[tryReuseRunner]: Reuse of runner ${runner.instanceId}: Tags deleted`);
+
+          await expBackOff(() => {
+            return metrics.trackRequestRegion(
+              runner.awsRegion,
+              metrics.ec2CreateReplaceRootVolumeTaskSuccess,
+              metrics.ec2CreateReplaceRootVolumeTaskFailure,
+              () => {
+                return ec2
+                  .createReplaceRootVolumeTask({
```

Comment:
Any idea how long this operation takes?

Reply:
no. I don't send metrics for it specifically on the spa.

Reply:
![Screenshot 2025-03-06 at 14 30 28](https://github.com/user-attachments/assets/476eae00-881c-4330-88de-90afdc562072)

arround 500ms, lets assume <1000ms